### ISSUE TITLE
add a method to get a summary of available backends

### ISSUE THREAD

**Issue by grlee77** [Quansight-Labs/uarray#180]
from the discussion in scipy/scipy#10383

>   should uarray have some sort of function to report what backends are currently global and/or registered?

**Comment by hameerabbasi** [comment]
@grlee77 Would you prefer this in the first version?

**Comment by grlee77** [comment]
I don't think it is critical, just something to consider potentially adding at some point.

**Comment by hameerabbasi** [comment]
cc @peterbell10 This is important for `dask.distributed`, not just getting, but setting global backend state.

I was thinking of the following approach:

- Transform the set of backends into an iterator of an internal type.
- Return the iterator of the type wrapped in Python when asked
- Provide a way to set the global state.

**Comment by peterbell10** [comment]
Can you explain what's different about `dask.distributed`?

**Comment by hameerabbasi** [comment]
The Python process is separate, possibly even on a different machine.

**Comment by peterbell10** [comment]
So the whole of uarray's state needs to be persisted into another process. In that case it would probably be easiest to just return something like a dictionary of lists, otherwise the pickling would have to be handled manually.

**Comment by hameerabbasi** [comment]
That would be good, yes. 😄 I'm wonder if there's a way to do this with `multiprocessing` that doesn't involve modifying end-user code.

**Comment by hameerabbasi** [comment]
Internally, we should keep C++ native objects, so as not to degrade performance.

**Comment by hameerabbasi** [comment]
Is this resolved by #244 and #220? Or would such a list still be preferred?
